apps/web/pages/merch/index.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Divider, Select, Heading, Grid } from "@chakra-ui/react";
+import { useQuery } from "@tanstack/react-query";
+import { Card, MerchListSkeleton, Page } from "ui/components/merch";
+import { QueryKeys } from "features/merch/constants";
+import { api } from "features/merch/services/api";
+import { Product } from "types";
+import { isOutOfStock } from "features/merch/functions";
+
+const MerchandiseList = () => {
+  const [selectedCategory, setSelectedCategory] = useState<string>("");
+  const [isMerchDisabled, setIsMerchDisabled] = useState<boolean | null>(false);
+  const [disabledText, setDisabledText] = useState<string>("");
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const fetchMerchSaleStatus = async () => {
+      try {
+        // TODO: change to use query?
+        const { disabled, displayText } = await api.getMerchSaleStatus();
+        setDisabledText(displayText ?? "");
+        setIsMerchDisabled(disabled);
+        setLoading(false);
+      } catch (error) {
+        // TODO: display error
+        setLoading(false);
+      }
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    fetchMerchSaleStatus();
+  }, []);
```

Comment:
@chanbakjsd Please evaluate if we should use useQuery for this fetch

Reply:
Yes, please use `useQuery` for this if possible as it is an API call.